src/main/java/me/zeroeightsix/fiber/tree/ConfigNodeImpl.java

```diff
@@ -50,6 +56,18 @@ public ConfigBranch getParent() {
         return this.parent;
     }
 
+    @Override
+    public Map<Identifier, ConfigAttribute<?>> getAttributes() {
+        return this.attributes;
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public <A> ConfigAttribute<A> getOrCreateAttribute(Identifier id, @Nonnull A defaultValue) {
+        return (ConfigAttribute<A>) getAttributes()
+                .computeIfAbsent(id, i -> new ConfigAttributeImpl<>(defaultValue, (Class<A>) defaultValue.getClass()));
+    }
```

Comment:
Careful, if the default value is of a subtype this won't work (hence the unchecked warning).

Reply:
oh no, how could I forget about that